### DISCUSSION TITLE
Consider a reactive paradigm

### DISCUSSION THREAD

**Discussion by matt-goldman** [SSWConsulting/SSW.Rewards.Mobile#474] [Ideas]
cc @AntPolkanov @bradystroud @JackDevAU @christoment 

The SSW Rewards app seems quite straightforward but there's actually quite a lot of moving parts. User state for example is updated from multiple places throughout the app.

We have at least 13 instances of `WeakReferenceMessenger.Send([...])` throughout the app, which in my mind is always a strong indicator that a reactive approach should be considered.

MVVM is largely working well so we don't necessarily need to scrap it in favour of MVU (in fact this would be quite hard to achieve without rewriting the UI in C# markup), but we could incorporate ReactiveUI and consider some centralised state management.

Off the top of my head, things that would benefit from this:

* Changing the user's profile pic
* Changing the user's points, balance etc
* Changing the state of achievements on the achievement grid
* The way that we currently display the achievement snackbar is bad (we should refactor this anyway, even if we don't use this approach)

It will become more useful once we incorporate some of the other items on the backlog, most notably notifications, but also when we unlock quizzes. We may also introduce more RTC functionality for conferences and events.

Even without the new features, I think it will simplify our code and make maintenance and troubleshooting easier.

**Comment by JackDevAU** [comment]
I'm all for it!

**Comment by matt-goldman-ssw** [comment]
Closing this discussion as @zacharykeeping has already started implementing this.
